game-core/src/main/java/games/strategy/engine/lobby/client/ui/GameListModel.java

```diff
@@ -0,0 +1,76 @@
+package games.strategy.engine.lobby.client.ui;
+
+import com.google.common.base.Preconditions;
+import games.strategy.net.GUID;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.triplea.lobby.common.GameDescription;
+import org.triplea.util.Tuple;
+
+class GameListModel {
```

Comment:
The code in this class is extracted from LobbyGameTableModel.

One notable update is it has a 'containsGame' API now, the table model has some small logic to do a conditional check if a game exists and then it calls 'add' or 'update' as appropriate.